### PR TITLE
Ticket4047 refl scripts

### PR DESCRIPTION
**Pull request by John-Holt-Tessella**
For ticket https://github.com/ISISComputingGroup/IBEX/issues/4047

### CODE REVIEW

File: techniques/reflectometry/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+import base
+import sample
```
Comment (rprospero) [inline techniques/reflectometry/__init__.py]: I'd change this to

from . import base, sample

This will future proof us for Python 3.

**Comment by John-Holt-Tessella** [comment]
1. Done
2. Interesting, I will look at this
3. Yes recording it as a doc test is really good I will do this.
4. Ok sorry you did not get the latest code I got it mixed up
5. OK I will do that before this is approved.

File: technique/reflectometry/base.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+        footprint_at_theta = footprint * sin(radians(theta))
+        s1 = 2 * s1sa * tan(radians(resolution * theta)) - footprint_at_theta
+        s2 = (constants.s1s2 * (footprint_at_theta + s1) / s1sa) - s1
+        return s1, s2
+
+    def _set_h_gaps(self, s1hg, s2hg, s3hg, s4hg):
+        print("Setting hgaps to {} (None's are not changed)".format([s1hg, s2hg, s3hg, s4hg]))
+        if not self.dry_run:
+            if s1hg is not None:
+                g.cset("S1HG", s1hg)
```
Comment (John-Holt-Tessella) [inline technique/reflectometry/base.py]: Consider adding `dryrun` to cset is some way. Discussion is needed maybe make a ticket.

**Comment by John-Holt-Tessella** [comment]
OK, will leave doc tests until it is running on instrument well

File: technique/reflectometry/base.py [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+        count_frames: the number of frames to wait for; None for don't count
+        s1vg: slit 1 vertical gap; None to use sample footprint and resolution
+        s2vg: slit 2 vertical gap; None to use sample footprint and resolution
+        s3vg: slit 3 vertical gap; None use fraction of maximum based on theta
+        s4vg: slit 4 vertical gap; None use fraction of maximum based on theta
+        smangle: super mirror angle, place in the beam, if set to 0 remove from the beam; None don't move super mirror
+        mode: mode to run in; None don't change modes
+        auto_height: if True when taking data run the auto-height routine
+        dry_run: True just print what is going to happen; False do the experiment
+
```
Comment (John-Holt-Tessella) [inline technique/reflectometry/base.py]: Add example

File: technique/reflectometry/base.py [hunk 4]
```diff
@@ -0,0 +1,503 @@
...
+        print("Setup only no measurement")
+    else:
+        movement.count_for(count_uamps, count_seconds, count_frames)
+
+
+def transmission(sample, title, s1vg, s2vg, s3vg=None, s4vg=None,
+                 count_seconds=None, count_uamps=None, count_frames=None,
+                 s1hg=None, s2hg=None, s3hg=None, s4hg=None,
+                 height_offset=5, smangle=None, mode=None, dry_run=False):
+    """
```
Comment (John-Holt-Tessella) [inline technique/reflectometry/base.py]: add example

**Pull request merged**